tests/test_base.py

```diff
@@ -1043,3 +1050,52 @@ class TestSchema(Schema):
     assert side_effects == [1]
     TestSchema({'lazy_side_effect': 1})
     assert side_effects == [1]
+
+
+def test_uuid_field():
+    my_id = '1a4c0351-3621-4860-b528-5ada0003fbda'
+
+    with pytest.raises(ValidationError):
+        UUID().clean('x')
+
+    with pytest.raises(ValidationError):
+        assert UUID().clean(None)
+
+    assert UUID().clean(my_id) == PythonUUID(my_id)
+    assert UUID().serialize(PythonUUID(my_id)) == my_id
+
+
+def test_clean_dict():
+    f = CleanDict(key_schema=Regex(r'^k*$'), value_schema=Regex(r'^v*$'))
+    assert f.clean({'k': 'v'}) == {'k': 'v'}
+    assert f.clean({'kkk': 'vvv', 'k': 'v'}) == {'kkk': 'vvv', 'k': 'v'}
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 'v'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'k': 'a'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 'a'})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': 1})
+
+    with pytest.raises(ValidationError):
+        f.clean({'a': [1]})
+
+    class TestSchema(Schema):
+        f_opt = CleanDict(
+            key_schema=Regex(r'^k*$'), value_schema=Integer(), required=False
+        )
+
+    assert TestSchema({'f_opt': {}}).full_clean() == {'f_opt': None}
+    assert TestSchema({'f_opt': {'k': 1}}).full_clean() == {'f_opt': {'k': 1}}
```

Comment:
Maybe we can also try cleaning a value of `'1'` (str) into `1` (int)?

Reply:
That would test `Integer`'s implementation. My goal here is to have anything with a clear match/not match separation.